### ISSUE TITLE
GetCookies cannot fetch cookies from IE 10 or IE 11 using IEDriverServer 32/64 bit

### ISSUE THREAD

**Issue by Prabhs13** [SeleniumHQ/selenium#1561]
Hi Guys,

I tried to get cookies from Internet Explorer Browser version 10 and 11 on Windows 8 (64-bit) by using IEDriverServer_2.50.0 ( both 32bit and 64 bit versions) by method "driver.manage().getCookies()", but was unable to get it and then the browser window hangs and I had to forcefully close it. Though it works fine using the IEDriverServer v2.46.

These are the logs from IEDriverServer v2.46 (32-bit)

DEBUG [PoolService-0] (java:608) - File download for Internet Explorer browser
 INFO [PoolService-0] (java:624) - absoluteFilePath: \test\resources\testData\bma04.xlsx
 INFO [PoolService-0] (java:626) - absoluteFilePath: \test\resources\testData\bma04.xlsx
DEBUG [PoolService-0] (java:499) - Session Cookie: SESSION="TPI::urn:sdf:xyz.com/xl/pqrs:user/21153474::4ddf884a-d748-4195::1454062666028::lHpK4xmE74VR7xHjxTkcuRLc+Cw="; path=/
DEBUG [PoolService-0] (java:500) - All Cookies: [_gat=1; path=/, _ga=GA1.2.2079055963; path=/, SESSION="TPI::urn:sdf:xyz.com/xl/pqrs:user/21153474::4ddf884a-d748-4195::1454062666028::lFpK4xmE74VR7XHjPTkcuRLc+Cw="; path=/]

**Comment by jimevans** [comment]
Cookie handling in the IE driver was rewritten in 2.47.0. Those lines don't look like logs from IEDriverServer. Additionally, the log information from 2.50.0 of IEDriverServer.exe up until the time it hangs would be useful.

**Comment by Prabhs13** [comment]
Hi @jimevans ,

Below attached are the IEDriverServer logs of INFO and DEBUG levels for the IE Driver versions 2.46.0 and 2.50.0


[IEDriverServerLogs_INFO_2.50.txt](https://github.com/SeleniumHQ/selenium/files/115535/IEDriverServerLogs_INFO_2.50.txt)
[IEDriverServerLogs_INFO_2.46.txt](https://github.com/SeleniumHQ/selenium/files/115536/IEDriverServerLogs_INFO_2.46.txt)

**Comment by jimevans** [comment]
Please provide logs at the `TRACE` level.

**Comment by Prabhs13** [comment]
Sure,

Hi @jimevans ,

Below are the logs of Trace level.

[IEDriverServerLogs_TRACE_2.50.txt](https://github.com/SeleniumHQ/selenium/files/117029/IEDriverServerLogs_TRACE_2.50.txt)

**Comment by jimevans** [comment]
That's helpful. Now, as the next troubleshooting step, can you please try this with a simple repro case against a public site. It need not be a complicated piece of WebDriver code; simply navigating to a site and getting the cookies should be sufficient.

As a side note, retrieving cookies is not globally broken, even for IE 10 or 11, and even on Windows 8.1. As proof, we have tests in the WebDriver project test suite for that scenario, and our CI system runs them against those versions without fail.

**Comment by Prabhs13** [comment]
Hi @jimevans ,

I would work on it from Monday..and will report its TRACE level logs.

**Comment by Prabhs13** [comment]
Hi @jimevans 

I have tried the suggested scenario on a public site and your code for "get cookie" works. But I still can’t figure it out, why cookie cannot be retrieved on our site using IE Driver versions 2.50.0.0 though it is possible with version 2.46.0.0. as shown in provided logs. The last message when the IE browser hangs using IE Driver V2.50.0.0 shown is : Entering CookieManager::ThreadProc..
Any clues??
Otherwise you may close this issue.
Thank you

**Comment by kriegaex** [comment]
We have the same problem here on one specific 3rd party HTTPS page with one PHP session cookie. I can easily get the cookies  via JavaScript, but not via getCookies in WebDriver. I can easily reproduce it, but only on our internal test system in a test via TestChameleon. Locally it works. My Selenium version is 2.53.

**Comment by kriegaex** [comment]
Another question: https://github.com/SeleniumHQ/selenium/blob/master/cpp/iedriverserver/CHANGELOG#L68 says:

> Removing IE driver support for ie.validateCookieDocumentType capability.
   This capability was made obsolete with the rewrite of the IE driver's
   cookie handling in 2.47. It has been ignored since, so this commit removes
   the IE driver code for that capability.

Is this somehow related to the problem? I noticed that on the page where I cannot read cookies I also cannot set one, maybe due to the "no-cache" header. Just guessing. I found something here:
https://github.com/SeleniumHQ/selenium-google-code-issue-archive/issues/1227
